Intermittent UT failure - `TestGetTransactionByIDCached`

https://github.com/hyperledger/firefly/runs/5618389248?check_suite_focus=true

```
?   	github.com/hyperledger/firefly/internal/tokens/tifactory	[no test files]
time="2022-03-20T16:33:07Z" level=error msg="Namespace mismatch for transaction 'cc30a0ca-e3bb-41b9-8234-9c436dc1011b' existing=ns2 new=ns1"
time="2022-03-20T16:33:07Z" level=error msg="Type mismatch for transaction 'c61f484a-e476-4aa1-bd40-8a7bff82ac14' existing=contract_invoke new=batch_pin"
--- FAIL: TestGetTransactionByIDCached (0.00s)
panic: 
assert: mock: The method has been called over 1 times.
	Either do one more Mock.On("GetTransactionByID").Return(...), or remove extra call.
	This call was unexpected:
		GetTransactionByID(*context.emptyCtx,*fftypes.UUID)
		0: (*context.emptyCtx)(0xc00009e010)
		1: &fftypes.UUID{0xab, 0xca, 0x46, 0xd0, 0xa, 0xa7, 0x43, 0xf0, 0xb2, 0xc2, 0xe7, 0xe0, 0x9, 0x63, 0xe4, 0x3f}
	at: [plugin.go:2069 txcommon.go:70 txcommon_test.go:416] [recovered]
	panic: 
assert: mock: The method has been called over 1 times.
	Either do one more Mock.On("GetTransactionByID").Return(...), or remove extra call.
	This call was unexpected:
		GetTransactionByID(*context.emptyCtx,*fftypes.UUID)
		0: (*context.emptyCtx)(0xc00009e010)
		1: &fftypes.UUID{0xab, 0xca, 0x46, 0xd0, 0xa, 0xa7, 0x43, 0xf0, 0xb2, 0xc2, 0xe7, 0xe0, 0x9, 0x63, 0xe4, 0x3f}
	at: [plugin.go:2069 txcommon.go:70 txcommon_test.go:416]

goroutine 76 [running]:
testing.tRunner.func1.2(0x79c100, 0xc000425c20)
	/opt/hostedtoolcache/go/1.16.14/x64/src/testing/testing.go:1153 +0x332
testing.tRunner.func1(0xc000466480)
	/opt/hostedtoolcache/go/1.16.14/x64/src/testing/testing.go:1156 +0x4b6
panic(0x79c100, 0xc000425c20)
	/opt/hostedtoolcache/go/1.16.14/x64/src/runtime/panic.go:965 +0x1b9
github.com/stretchr/testify/mock.(*Mock).fail(0xc0003f5b80, 0x838969, 0xa5, 0xc000453140, 0x4, 0x4)
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.7.0/mock/mock.go:254 +0x17f
github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc0003f5b80, 0x8f81bd, 0x12, 0xc000460b60, 0x2, 0x2, 0xc000452fc0, 0x4, 0x4)
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.7.0/mock/mock.go:399 +0x2ee
github.com/stretchr/testify/mock.(*Mock).Called(0xc0003f5b80, 0xc000460b60, 0x2, 0x2, 0xc0000ade4c, 0xc0004[609](https://github.com/hyperledger/firefly/runs/5618389248?check_suite_focus=true#step:4:609)a0, 0x8490e0)
	/home/runner/go/pkg/mod/github.com/stretchr/testify@v1.7.0/mock/mock.go:383 +0x1ab
github.com/hyperledger/firefly/mocks/databasemocks.(*Plugin).GetTransactionByID(0xc0003f5b80, 0x8a2970, 0xc00009e010, 0xc0004013a0, 0xc0003f5c70, 0x0, 0x0)
	/home/runner/work/firefly/firefly/mocks/databasemocks/plugin.go:2069 +0xb3
github.com/hyperledger/firefly/internal/txcommon.(*transactionHelper).GetTransactionByIDCached(0xc000455800, 0x8a2970, 0xc00009e010, 0xc0004013a0, 0x7e6e00, 0xc0004013a0, 0x0)
	/home/runner/work/firefly/firefly/internal/txcommon/txcommon.go:70 +0x10a
github.com/hyperledger/firefly/internal/txcommon.TestGetTransactionByIDCached(0xc000466480)
	/home/runner/work/firefly/firefly/internal/txcommon/txcommon_test.go:416 +0x3bf
testing.tRunner(0xc000466480, 0x848148)
	/opt/hostedtoolcache/go/1.16.14/x64/src/testing/testing.go:1203 +0xe5
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.16.14/x64/src/testing/testing.go:1248 +0x2b3
```

--- a/internal/txcommon/txcommon_test.go
+++ b/internal/txcommon/txcommon_test.go
@@ -20,7 +20,6 @@ import (
 	"context"
 	"fmt"
 	"testing"
-	"time"
 
 	"github.com/hyperledger/firefly-common/pkg/config"
 	"github.com/hyperledger/firefly-common/pkg/fftypes"
@@ -376,7 +375,7 @@ func TestGetTransactionByIDCached(t *testing.T) {
 
 	mdi := &databasemocks.Plugin{}
 	mdm := &datamocks.Manager{}
-	txHelper, txCache, _ := NewTestTransactionHelper(mdi, mdm)
+	txHelper, _, _ := NewTestTransactionHelper(mdi, mdm)
 	ctx := context.Background()
 
 	txid := fftypes.NewUUID()
@@ -388,15 +387,9 @@ func TestGetTransactionByIDCached(t *testing.T) {
 		BlockchainIDs: core.FFStringArray{"0x111111"},
 	}, nil).Once()
 
-	previousTxCacheSize := txCache.ItemCount()
-
 	tx, err := txHelper.GetTransactionByIDCached(ctx, txid)
 	assert.NoError(t, err)
 	assert.Equal(t, txid, tx.ID)
-
-	for txCache.ItemCount() <= previousTxCacheSize {
-		time.Sleep(time.Millisecond * 1)
-	}
 
 	tx, err = txHelper.GetTransactionByIDCached(ctx, txid)
 	assert.NoError(t, err)

